parley/src/layout/editor.rs

```diff
@@ -21,6 +21,12 @@ pub enum ActiveText<'a> {
     Selection(&'a str),
 }
 
+#[derive(Clone, Default)]
+struct ComposeState {
+    selection: Option<(usize, usize)>,
+    text: Arc<str>,
```

Comment:
I feel like this should be a `String`. Then:
- We could mutate it and re-use the allocation when updating the pre-edit
- Setting it to empty would be allocation free

Reply:
> * Setting it to empty would be allocation free

it may not matter, but it's never empty